### PR TITLE
Category component, weekly stats, service

### PR DESCRIPTION
**Pull request by MaxFangX**
- Create CategoryList component
- Create CategoryService
- No scope soup (: https://toddmotto.com/no-scope-soup-bind-to-controller-angularjs/
- Weekly ColorCategories

### CODE REVIEW

**Comment by MaxFangX** [comment]
@cindytung no

**Comment by MaxFangX** [comment]
<3

File: cal/cal/static/js/app.js [hunk 1]
```diff
@@ -194,91 +192,102 @@ analyticsApp.component('tagDetails', {
...
+      return false;
+    } else {
+      return true;
+    }
+  }.bind(this);
+
+  this.create = function(category) {
+    // TODO this was never implemented
+    /* jshint unused:vars */
+  };
```
Comment (jamesuejio) [inline cal/cal/static/js/app.js]: I thought we weren't going to be able to create categories. Can just remove this no? Might confuse people in the future.

File: cal/cal/static/js/services.js [hunk 2]
```diff
@@ -127,3 +127,95 @@ analyticsApp.service("TagService", ['$http', function($http) {
...
+    }, function errorCallback(response) {
+      /* jshint unused:vars */
+      console.log("Failed to get categories");
+    });
+  };
+
+  this.createCategory = function(label, keywords) {
+    /* jshint unused:vars */
+    // TODO this was never implemented
+  };
```
Comment (jamesuejio) [inline cal/cal/static/js/services.js]: See above comment (we won't use create)

File: cal/api/views.py [hunk 3]
```diff
@@ -224,14 +224,6 @@ def get_serializer_context(self):
 
     def get_queryset(self):
         qs = Tag.objects.filter(user=self.request.user)
-        start = self.request.query_params.get('start')
-        end = self.request.query_params.get('end')
-
-        if start:
-            qs = qs.filter(start__gte=start)
-        if end:
-            qs = qs.filter(end__lte=end)
```
Comment (jamesuejio) [inline cal/api/views.py:233]: Why did you delete this again?

File: cal/cal/static/js/services.js [hunk 4]
```diff
@@ -127,3 +127,95 @@ analyticsApp.service("TagService", ['$http', function($http) {
...
+      }
+    }).then(function successCallback(response) {
+      _this.categories[timeRange] = [];
+      for (var i = 0; i < response.data.results.length; i++) {
+        var category = response.data.results[i];
+        _this.categories[timeRange].push({
+          id: category.id,
+          label: category.label,
+          hours: category.hours,
+          include: true
```
Comment (jamesuejio) [inline cal/cal/static/js/services.js]: WHO GIVES A F*** ABOUT AN OXFORD COMMA https://www.youtube.com/watch?v=P_i1xk07o4g
Add a comma after include: true

**Review by jamesuejio — CHANGES_REQUESTED** [review]
categories - cumulative preloader seems to be stuck. nevermind it works

File: cal/cal/static/js/app.js (hunk 1 above)
Comment (MaxFangX) [reply cal/cal/static/js/app.js]: done

File: cal/cal/static/js/services.js (hunk 4 above)
Comment (MaxFangX) [reply cal/cal/static/js/services.js]: done

File: cal/cal/static/js/services.js (hunk 2 above)
Comment (MaxFangX) [reply cal/cal/static/js/services.js]: done

File: cal/api/views.py (hunk 3 above)
Comment (MaxFangX) [reply cal/api/views.py:233]: This breaks it - a `Tag` doesn't have a start parameter so it causes a 500 error

Comment (jamesuejio) [reply cal/api/views.py:233]: Okay

**Review by jamesuejio — APPROVED** [review]

**Pull request merged**